core/mavlink_system.cpp

```diff
@@ -282,16 +295,82 @@ void Device::device_thread(Device *self)
     }
 }
 
-void Device::send_heartbeat(Device &self)
+std::string MAVLinkSystem::component_name(uint8_t component_id)
+{
+    switch (component_id) {
+        case MAV_COMP_ID_AUTOPILOT1:
+            return "Autopilot";
+        case MAV_COMP_ID_CAMERA:
+            return "Camera 1";
+        case MAV_COMP_ID_CAMERA2:
+            return "Camera 2";
+        case MAV_COMP_ID_CAMERA3:
+            return "Camera 3";
+        case MAV_COMP_ID_CAMERA4:
+            return "Camera 4";
+        case MAV_COMP_ID_CAMERA5:
+            return "Camera 5";
+        case MAV_COMP_ID_CAMERA6:
+            return "Camera 6";
+        case MAV_COMP_ID_GIMBAL:
+            return "Gimbal";
+        default:
+            return "Unsupported component";
+    }
+}
+
+void MAVLinkSystem::add_new_component(uint8_t component_id)
+{
+    auto res_pair = _components.insert(component_id);
+    if (res_pair.second) {
+        LogDebug() << "Component " << component_name(component_id) << " added.";
+    }
+}
+
+size_t MAVLinkSystem::total_components() const
+{
+    return _components.size();
+}
+
+bool MAVLinkSystem::is_standalone() const
+{
+    return !is_autopilot();
+}
+
+bool MAVLinkSystem::is_autopilot() const
+{
+    return get_autopilot_id() != uint8_t(0);
+}
+
+bool MAVLinkSystem::has_camera(uint8_t camera_id) const
+{
+    uint8_t camera_comp_id = MAV_COMP_ID_CAMERA + (camera_id - 1);
+
+    for (auto compid : _components) {
+        return compid == camera_comp_id;
```

Comment:
I think this is wrong and should be:
```
for (auto compid : _components) {
    if (compid == camera_comp_id) {
        return true;
    }
}
```

Reply:
No its not. The result of the expression `compid == camera_comp_id` evaluates to `true` if they're equal and `false` otherwise.

Reply:
Oh! ok sorry I missed the point. I'll correct.